### PR TITLE
Upgrade react router

### PR DESCRIPTION
**Pull request by anders-hopland**
React on-enter is deprecated, upgraded to new render method in v4. Also changed joinpage link from a tag to link tag

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -69,8 +69,9 @@ export class App extends Component {
     document.querySelector('body').scrollTop = 0
     document.querySelector('#app').scrollTop = 0
 
-    const [first, ...rest] = args.location.pathname.slice(1) || 'home'
-    document.title = 'Ascend NTNU - ' + first.toUpperCase() + rest
+    //const [first, ...rest] = args.location.pathname.slice(1) || 'home'
+    //document.title = 'Ascend NTNU - ' + first.toUpperCase() + rest
+    console.log("Changing")
```
Comment (niklasmh) [inline src/App.js]: Maybe remove debug log

File: src/App.js [hunk 2]
```diff
@@ -87,37 +88,101 @@ export class App extends Component {
...
         <Switch>
-          <Route exact path="/" component={FrontPage} onEnter={this.gotoTop} />
-          <Route path="/about" component={AboutPage} onEnter={this.gotoTop} />
-          <Route path="/blog" component={BlogPage} onEnter={this.gotoTop} />
-          <Route path="/blog/tags/:tags?" component={BlogPage} />
-          <Route path="/blog/:post" component={BlogPage} />
+          <Route exact path="/" 
+            render={(props) => {
+              this.gotoTop()
+              console.log("testing")
```
Comment (niklasmh) [inline src/App.js]: And some other debug logs

File: src/App.js [hunk 3]
```diff
@@ -87,37 +88,101 @@ export class App extends Component {
...
+              return <FrontPage {...props}/>
+            }} 
+          />
+          <Route 
+            path="/about" 
+            render={(props) => {
+              this.gotoTop()
+              console.log("testing")
+              return <AboutPage {...props}/>
+            }}
```
Comment (niklasmh) [inline src/App.js]: What about making a function that has component (here AboutPage) as parameter, then use `render={(props) => this.switchPage(AboutPage, props)}`

File: src/App.js (hunk 2 above)
Comment (anders-hopland) [reply src/App.js]: Fixed

File: src/App.js (hunk 1 above)
Comment (anders-hopland) [reply src/App.js]: Fixed

File: src/App.js [hunk 4]
```diff
@@ -69,8 +69,8 @@ export class App extends Component {
     document.querySelector('body').scrollTop = 0
     document.querySelector('#app').scrollTop = 0
 
-    const [first, ...rest] = args.location.pathname.slice(1) || 'home'
-    document.title = 'Ascend NTNU - ' + first.toUpperCase() + rest
+    //const [first, ...rest] = args.location.pathname.slice(1) || 'home'
+    //document.title = 'Ascend NTNU - ' + first.toUpperCase() + rest
```
Comment (niklasmh) [inline src/App.js]: Doesn't this work? It only changes the title in the tab

Comment (anders-hopland) [reply src/App.js]: It doesn't work, thought to do that afterwards. Could just fix it in this pull request as well

Comment (niklasmh) [reply src/App.js]: Is it possible to make it work? Seems like the problem is the `args` provided. It is not essential though.

Comment (anders-hopland) [reply src/App.js]: could pass in the wanted text as an argument

Comment (niklasmh) [reply src/App.js]: Maybe just add an option in the this.switchPage(...., title) such that it can be provided manually

**Pull request merged**